src/ray/object_manager/memory_object_reader.h

```diff
@@ -0,0 +1,43 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "ray/object_manager/plasma/client.h"
+#include "ray/object_manager/spilled_object.h"
+
+namespace ray {
+
+/// A wrapper over plasma object.
+/// This class is thread safe.
+class MemoryObjectReader : public IObjectReader {
```

Comment:
Why don't we just call it IRayObject or sth?

Reply:
too many Objects...

Reply:
Hmm I see lol. Then can we change SpillObject -> SpillObjectReader (just for the consistency)?

Reply:
yup. I'm trying to keep this RP as small as possible...